matchtree_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zoekt
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_breakOnNewlines(t *testing.T) {
+	type args struct {
+		cm   *candidateMatch
+		text []byte
+	}
+	tests := []struct {
+		name string
+		args args
+		want []*candidateMatch
+	}{
+		{
+			name: "trivial case",
+			args: args{
+				cm: &candidateMatch{
+					byteOffset:  0,
+					byteMatchSz: 0,
+				},
+				text: nil,
+			},
+			want: nil,
+		},
+		{
+			name: "no newlines",
+			args: args{
+				cm: &candidateMatch{
+					byteOffset:  0,
+					byteMatchSz: 1,
+				},
+				text: []byte("a"),
+			},
+			want: []*candidateMatch{
+				{
+					byteOffset:  0,
+					byteMatchSz: 1,
+				},
+			},
+		},
+		{
+			name: "newline at start",
+			args: args{
+				cm: &candidateMatch{
+					byteOffset:  0,
+					byteMatchSz: 2,
+				},
+				text: []byte("\na"),
+			},
+			want: []*candidateMatch{
+				{
+					byteOffset:  1,
+					byteMatchSz: 1,
+				},
+			},
+		},
+		{
+			name: "newline at end",
```

Comment:
Would be good to have two additional variants here:

- newline at start, 3 lines
- newline at end, 3 lines

i.e. multi-line variants of this, specifically because `breakOnNewlines` internally handles the 2nd iteration differently than the 1st.